src/contradictory_claims/data/make_dataset.py

```diff
@@ -27,19 +31,53 @@ def load_multi_nli(train_path: str, test_path: str):
                                         label in multinli_test_data.gold_label]
 
     # Insert the CLS and SEP tokens
-    x_train = '[CLS]' + multinli_train_data.sentence1 + '[SEP]' + multinli_train_data.sentence2
-    x_train = x_train.to_numpy()
-    x_test = '[CLS]' + multinli_test_data.sentence1 + '[SEP]' + multinli_test_data.sentence2
-    x_test = x_test.to_numpy()
-
-    # Reformat to one-hot categorical variable (3 columns)
-    y_train = np_utils.to_categorical(multinli_train_data.gold_label, dtype='int')
-    y_test = np_utils.to_categorical(multinli_test_data.gold_label, dtype='int')
+    if multi_class:
+        x_train = '[CLS]' + multinli_train_data.sentence1 + '[SEP]' + multinli_train_data.sentence2
+        x_train = x_train.to_numpy()
+        x_test = '[CLS]' + multinli_test_data.sentence1 + '[SEP]' + multinli_test_data.sentence2
+        x_test = x_test.to_numpy()
+
+        # Reformat to one-hot categorical variable (3 columns)
+        y_train = np_utils.to_categorical(multinli_train_data.gold_label, dtype='int')
+        y_test = np_utils.to_categorical(multinli_test_data.gold_label, dtype='int')
+    else:
+        # Add the category info (CON, ENT, NEU) as auxillary text at the end
+        x_train_1 = '[CLS]' + multinli_train_data.sentence1 + '[SEP]' + multinli_train_data.sentence2 +\
+                    '[SEP]' + 'CON'
+        x_train_2 = '[CLS]' + multinli_train_data.sentence1 + '[SEP]' + multinli_train_data.sentence2 +\
+                    '[SEP]' + 'ENT'
+        x_train_3 = '[CLS]' + multinli_train_data.sentence1 + '[SEP]' + multinli_train_data.sentence2 +\
+                    '[SEP]' + 'NEU'
+        x_train = x_train_1.append(x_train_2)
+        x_train = x_train.append(x_train_3)
+        x_train = x_train.to_numpy()
+        x_test_1 = '[CLS]' + multinli_test_data.sentence1 + '[SEP]' + multinli_test_data.sentence2 +\
+                   '[SEP]' + 'CON'
+        x_test_2 = '[CLS]' + multinli_test_data.sentence1 + '[SEP]' + multinli_test_data.sentence2 +\
+                   '[SEP]' + 'ENT'
+        x_test_3 = '[CLS]' + multinli_test_data.sentence1 + '[SEP]' + multinli_test_data.sentence2 +\
+                   '[SEP]' + 'NEU'
+        x_test = x_test_1.append(x_test_2)
+        x_test = x_test.append(x_test_3)
+        x_test = x_test.to_numpy()
```

Comment:
Feels like this functionality is copy-pasted and thus could be made more concise with helper functions. For the sake of merging now it's fine, but if we wanted to clean up and refactor in future, this is something to keep in mind

Reply:
Good point! Agree.